users/views.py

```diff
@@ -40,18 +38,17 @@ def post(self, request, *args, **kwargs):
 #     template_name = 'registration/signup.html'
 
 
-class UserDetail(generics.RetrieveAPIView):
-    queryset = UserProfile.objects.all()
+class MyProfileDetailView(generics.RetrieveAPIView):
     serializer_class = UserSerializer
-    permission_classes = (IsAuthenticated, IsOwner)
+    permission_classes = (IsAuthenticated,)
 
-    def get(self, request, *args, **kwargs):
-        token = request.META.get('HTTP_AUTHORIZATION')
-        token = token.replace('Token ', '')
-        user = Token.objects.get(key=token).user
-        return Response({
-            'token': token,
-            'username': user.username,
-            'email': user.email,
-        })
+    def get_object(self):
```

Comment:
You should check the permission in this method. You can read the original `get_object` method of `RetrieveAPIView`.